lib/stack_master/commands/apply.rb

```diff
@@ -85,14 +95,43 @@ def create_stack
         cf.create_stack(stack_options.merge(tags: proposed_stack.aws_tags))
       end
 
+      def upload_files
+        s3.upload_files(s3_options)
+      end
+
+      def template_method
+        return :template_body unless use_s3?
+        :template_url
+      end
+
+      def template_value
+        return proposed_stack.maybe_compressed_template_body unless use_s3?
+        s3.url(@s3_config.merge('template' => @stack_definition.template))
+      end
+
+      def files_to_upload
+        return [] unless use_s3?
+        [@stack_definition.files_to_upload, @stack_definition.template_file_path].flatten
```

Comment:
We wouldn't want to upload the raw `@stack_definition.template_file_path` for sparkleformation templates. I think we should use `proposed_stack.maybe_compressed_template_body` instead.